components/sessionReady/index.jsx

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/* eslint-disable no-console */
+import T from "prop-types";
+import React, { useEffect, useState, useCallback } from "react";
+import { useRouter } from "next/router";
+import { useSession } from "@inrupt/solid-ui-react";
+
+import Spinner from "../spinner";
+
+export default function SessionReady({ children }) {
+  const [sessionError, setSessionError] = useState();
+  const { session, sessionRequestInProgress } = useSession();
+  const router = useRouter();
+
+  const onError = useCallback(
+    (error) => {
+      setSessionError(error);
+      // FIXME: Remove once interaction_required bug in SDK is resolved:
+      session.logout();
```

Comment:
@daytonn @VirginiaBalseiro @NSeydoux I've added a temporary `session.logout()` here whilst we wait on the fix for the interaction_required bug; Sure "logging out" of the SDK isn't great, but it's not a logout from the IDP, it just prevents the retries from heavily looping.